core/sync.js

```diff
@@ -332,13 +336,12 @@ class Sync {
         // and especially avoid deep nesting levels.
         delete doc.moveFrom
         delete doc.overwrite
+        if (doc.deleted) doc._deleted = true
       }
 
       log.trace({ path, seq }, `Applied change on ${sideName} side`)
       await this.pouch.setLocalSeqAsync(change.seq)
-      if (!change.doc._deleted) {
-        await this.updateRevs(change.doc, sideName)
-      }
+      await this.updateRevs(doc, sideName)
```

Comment:
No migration phase is planned? I mean if a user deletes a file locally with the current version of the desktop client while offline, and migrates to the new version of the desktop client, the client won't try to synchronize this deletion. Right?

Reply:
Right, I need to think about this.